backend/src/main/scala/com/softwaremill/adopttapir/starter/api/StarterRequest.scala

```diff
@@ -13,113 +15,32 @@ case class StarterRequest(
-  case object Netty extends ServerImplementationRequest {
-    override def toModel: ServerImplementation = ServerImplementation.Netty
-  }
-
-  case object Http4s extends ServerImplementationRequest {
-    override def toModel: ServerImplementation = ServerImplementation.Http4s
-  }
-
-  case object ZIOHttp extends ServerImplementationRequest {
-    override def toModel: ServerImplementation = ServerImplementation.ZIOHttp
-  }
-
-  override def values: IndexedSeq[ServerImplementationRequest] = findValues
-}
-
-sealed trait JsonImplementationRequest extends EnumEntry {
-  def toModel: JsonImplementation
-}
-
-/** Changes in JSON implementation have to be reflected in .github/workflows/adopt-tapir-ci.yml file so that running jobs in parallel is
-  * still possible
-  */
-object JsonImplementationRequest extends Enum[JsonImplementationRequest] with CirceEnum[JsonImplementationRequest] {
-  case object No extends JsonImplementationRequest {
-    override def toModel: JsonImplementation = JsonImplementation.WithoutJson
-  }
-
-  case object Circe extends JsonImplementationRequest {
-    override def toModel: JsonImplementation = JsonImplementation.Circe
-  }
-
-  case object UPickle extends JsonImplementationRequest {
-    override def toModel: JsonImplementation = JsonImplementation.UPickle
-  }
-
-  case object Jsoniter extends JsonImplementationRequest {
-    override def toModel: JsonImplementation = JsonImplementation.Jsoniter
-  }
-
-  case object ZIOJson extends JsonImplementationRequest {
-    override def toModel: JsonImplementation = JsonImplementation.ZIOJson
-  }
-
-  override def values: IndexedSeq[JsonImplementationRequest] = findValues
-}
-
-sealed trait ScalaVersionRequest extends EnumEntry {
-  def toModel: ScalaVersion
-}
-
-/** Changes in Scala versions have to be reflected in .github/workflows/adopt-tapir-ci.yml file so that running jobs in parallel is still
-  * possible
-  */
-object ScalaVersionRequest extends Enum[ScalaVersionRequest] with CirceEnum[ScalaVersionRequest] {
-  case object Scala2 extends ScalaVersionRequest {
-    override def toModel: ScalaVersion = ScalaVersion.Scala2
-  }
-
-  case object Scala3 extends ScalaVersionRequest {
-    override def toModel: ScalaVersion = ScalaVersion.Scala3
-  }
-
-  override def values: IndexedSeq[ScalaVersionRequest] = findValues
-}
-
-sealed trait BuilderRequest extends EnumEntry {
-  def toModel: Builder
-}
-
-object BuilderRequest extends Enum[BuilderRequest] with CirceEnum[BuilderRequest] {
-  case object Sbt extends BuilderRequest {
-    override def toModel: Builder = Builder.Sbt
-  }
-
-  case object ScalaCli extends BuilderRequest {
-    override def toModel: Builder = Builder.ScalaCli
-  }
-
-  override def values: IndexedSeq[BuilderRequest] = findValues
-}
+) derives Decoder
+
+object StarterRequest:
+  inline given Encoder[StarterRequest] = deriveEncoder
+
+enum EffectRequest(val toModel: ServerEffect) derives JsonTaggedAdt.PureEncoder, JsonTaggedAdt.PureDecoder:
```

Comment:
maybe also add a comment why we need `JsonTaggedAdt` over the usual `derives Decoder`

Reply:
Added.